src/microsoft.sheets/spreadsheet_endpoint.bal

```diff
@@ -0,0 +1,579 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+import ballerina/oauth2;
+
+# Microsoft Spreadsheet Client Object.
+# + msSpreadsheetClient - HTTP client endpoint for the spreadsheet API
+# + microsoftGraphConfig - Configurations for accessing spreadsheet API
+public type MSSpreadsheetClient client object {
+    http:Client msSpreadsheetClient;
+    MicrosoftGraphConfiguration microsoftGraphConfig;
+
+    public function init(MicrosoftGraphConfiguration msGraphConfig) {
+        self.microsoftGraphConfig = msGraphConfig;
+        oauth2:OutboundOAuth2Provider oauth2Provider3 = new ({
+            accessToken: msGraphConfig.msInitialAccessToken,
+            refreshConfig: {
+                clientId: msGraphConfig.msClientID,
+                clientSecret: msGraphConfig.msClientSecret,
+                refreshToken: msGraphConfig.msRefreshToken,
+                refreshUrl: msGraphConfig.msRefreshURL,
+                clientConfig: {
+                    secureSocket: {
+                        trustStore: {
+                            path: msGraphConfig.trustStorePath,
+                            password: msGraphConfig.trustStorePassword
+                        }
+                    }
+                }
+            }
+        });
+        http:BearerAuthHandler oauth2Handler3 = new (oauth2Provider3);
+
+        self.msSpreadsheetClient = new (msGraphConfig.baseUrl, {
+                auth: {
+                    authHandler: oauth2Handler3
+                },
+                secureSocket: {
+                    trustStore: {
+                        path: msGraphConfig.trustStorePath,
+                        password: msGraphConfig.trustStorePassword
+                    }
+                }
+            });
+    }
+
+    # Open a Workbook by the given name.
+    # + path - Path to the workbook file
+    # + workbookName - Name of the Workbook
+    # + return - A Workbook client object on success, else returns an error
+    public remote function openWorkbook(string path, string workbookName) returns Workbook|Error {
+        Workbook workBook = new (self.microsoftGraphConfig, path, workbookName);
```

Comment:
Can't we pass the `msSpreadsheetClient`, we created earlier. I see the same client created inside Workbook

Reply:
It is possible to do so. Refactored the code to use the same HTTP client across all the client objects.